docs/REST/authentication.md

```diff
@@ -348,4 +348,47 @@ private static String buildContentHash(HttpUriRequest request) throws IOExceptio
 
     byte[] digest = new DigestUtils(SHA_256).digest(content);
     return Base64.getEncoder().encodeToString(digest);
-}
\ No newline at end of file
+}
+```
+### Golang
+```golang
+import (
+    "crypto/hmac"
+    "crypto/sha256"
+    "encoding/base64"
+    "fmt"
+    "io/ioutil"
+    "net/http"
+    "strings"
+    "time"
+)
+
+func getContentHashBase64(content string) string {
+    hasher := sha256.New()
+    hasher.Write([]byte(content))
+    return base64.StdEncoding.EncodeToString(hasher.Sum(nil))
+}
+
+func signRequest(content string, key []byte) string {
+    hmac := hmac.New(sha256.New, key)
+    hmac.Write([]byte(content))
+    return base64.StdEncoding.EncodeToString(hmac.Sum(nil))
+}
+
+// Setup the auth header for accessing Azure AppConfiguration service
+func PrepareAuthHeader(verb string, host string, pathAndQuery string, body string, id string, secret string, req *http.Request) {
```

Comment:
>verb string, host string, pathAndQuery string, body string [](start = 23, length = 58)

Is it common to pass around everything when working with HTTP request in go?

I would expect just passing in (req, id, secret) like in java or C#. Then getting req.Method, req.Host, req.URL, and req.GetBody() to prepare the authentication. With all the current parameters it kind of seems like the httprequest should be generated in the method rather than passed in.

Reply:
BTW, it will be easier to read to just call this function `Sign` or `SignRequest`. It will be consistent with other examples.

Reply:
Do you mind to change the `execute` in Java example to `SignRequest` too?